code/game/objects/items/devices/radio/beacon.dm

```diff
@@ -31,8 +39,8 @@
 	visible_message(SPAN_WARNING("\The [src] pops and cracks, and a thin wisp of dark smoke rises from the vents."), range = 2)
 	update_icon()
 	for(var/obj/machinery/computer/teleporter/T in SSmachines.machinery)
```

Comment:
Not asking for a change in this PR, but I feel like this would be better if the beacon kept a reference to anything locked onto it, rather than having to iterate the entire machinery list.

Reply:
This is something that's covered in my overhaul of teleporter beacons that'll be ported in a separate PR but depends on some of the changes in this one.